boot/boot.js

```diff
@@ -1491,7 +1490,10 @@ $tw.Wiki = function(options) {
 		$tw.utils.each(pluginTiddlers,function(tiddler) {
 			// Extract the constituent tiddlers
 			if($tw.utils.hop(pluginInfo,tiddler.fields.title)) {
-				$tw.utils.each(pluginInfo[tiddler.fields.title].tiddlers,function(constituentTiddler,constituentTitle) {
+				var constituentTiddlers = pluginInfo[tiddler.fields.title].tiddlers;
+				$tw.utils.each(constituentTiddlers,function(constituentTiddler,constituentTitle) {
+					// Skip translation tiddlers of plugin that follows `$:/plugins/xxx/yyy/languages/` pattern, will handle it later
```

Comment:
The dependency on the format of plugin names is very brittle.

An option we should consider is to extend the plugin format to include languages in additional properties to the existing "tiddlers" property:

```
{
	"tiddlers": {
		"$:/config/DefaultColourMappings/menubar-foreground": {
			"title": "$:/config/DefaultColourMappings/menubar-foreground",
			"text": "#fff"
		},
		"$:/config/DefaultColourMappings/menubar-background": {
....
	"languages": {
		"en-GB": {
			"$:/plugins/tiddlywiki/menubar/readme": {
				"title": "$:/plugins/tiddlywiki/menubar/readme",
				"text": "This is English"
			}
		},
		"fr-FR": {
			"$:/plugins/tiddlywiki/menubar/readme": {
				"title": "$:/plugins/tiddlywiki/menubar/readme",
				"text": "This is French"
			}
		}
	}
```

On the Node.js side there would need to be some reorganisation so that the language tiddlers can be distinguished from the ordinary plugin tiddlers. One approach might be to allow plugins to opt-in to a new folder format where there is are "tiddlers" and "languages" folders within the plugin folder. Perhaps

Reply:
I was reading science fiction novel A Deepness in the Sky last night, didn't saw this. I will put more time here in next few days.

> One approach might be to allow plugins to opt-in to a new folder format where there is are "tiddlers" and "languages" folders within the plugin folder. Perhaps

You mean when dev plugins? But recently we mostly use this folder structure https://github.com/tiddly-gittly/Modern.TiddlyDev/tree/master/src/plugin-name and plugins are published as JSON, even used in NodeJS wiki (So normal user moved here from Obsidian or Notion don't need to care about folder when installing plugin, as they used to). I always discourage manually using plugin folder in NodeJS wiki, for better new user experience.

I agree with adding a new `"languages":` field in JSON plugin, but how do we know it belongs to there when packaging plugin JSON? By a field or file type? Then we can simply use this information in boot.js:1496.

I suggest add a new tiddler type `text/vnd.tiddlywiki-language` or `text/vnd.tiddlywiki-localization` or `text/vnd.tiddlywiki-l10n`,

```multid
title: $:/language/
type: text/vnd.tiddlywiki-localization

CloseAll/Button: close all
ColourPicker/Recent: Recent:
```

Reply:
Oh, no, they can't have same title, otherwise they will overwrite each other in `constituentTiddlers`.

Reply:
Or we can remove the "skip" here, it only reduce the count of shadow tiddlers. It doesn't matter if we still load them.